### ISSUE TITLE
Requiring assets from node?

### ISSUE THREAD

**Issue by schovi** [topheman/react-es6-isomorphic#3]
Hi,

I am looking around for your great implementation of node+react and getting inspiration for my projects :)
There is one thing i am fighting with and you dont use that too. It is requiring of assets from webpack, but rendered on server side. You solved it like me by requiring assets only from stylesheets, which are completly outside Node.
Have you tried to solve that or is it completely impossible?

**Comment by topheman** [comment]
Hie @schovi 

I came accross the same kind of problem : the `resolve.alias` option of webpack. Checkout this part of the blog post I wrote about it : http://dev.topheman.com/react-es6-isomorphic/#react-es6-isomorphic-feedbacks-nodejs-webpack

TL;DR : Using webpack can be great but when it comes to use it in both worlds : node and browsers, you have to be careful with all the overwhelming options you can set. In that case, I finally use webPack aliases only browser side to load the config scripts but in the code, I used dependency injection (which would work in both browser and node).

FYI this is where I made the switch : https://github.com/topheman/react-es6-isomorphic/commit/6f78bdc64a6926c497570ac31c23bc99e8063b61 (it's about mocking a service making http requests - having different behaviors in dev/test/prod). If you looked into this part of the code (and you know webpack, you'll now understand why it's written like that ;-) )

I assume you have now your answer to your question reading my [blog post](http://dev.topheman.com/react-es6-isomorphic/) : it would mean overring node's `require()` ...

_Edit: This project is almost 4 months old, maybe things have changed (but not that I am aware of)_

**Comment by schovi** [comment]
@topheman thanks for answer. Haven't much time now, but i tried some digging and found maybe proof solution.

With using of npm package [app-module-path](https://www.npmjs.com/package/app-module-path). Usage [here](https://gist.github.com/branneman/8048520).

1) I can add in project/index.js file

```
import addModulePath from 'app-module-path';
addModulePath.addPath(__dirname + '/assets');
```

2) This allow you to resolve file in node with `require('myimage.png')`. This normally works in webpack, but node wont find that file.

3) You have to make custom require script

```
require.extensions['.' + ext] = (require) => {
  require.exports = `HEY its a .${ext}`
};
```

4) TODO :) IDEA: There I will load `stats.json` generated by webpack and I can find how `myimage.png` was resolved (url, or base64).

Will try to finish step 4 and publish result.
